internals/overlord/state/identities.go

```diff
@@ -266,17 +320,44 @@ func (s *State) Identities() map[string]*Identity {
 
 // IdentityFromInputs returns an identity with the given inputs, or nil
 // if there is none.
-func (s *State) IdentityFromInputs(userID *uint32) *Identity {
+//
+// Identity priority:
+//  1. If both username and password are provided, the function attempts to
+//     match a "basic" type identity. The userID is ignored in this case. If
+//     a matching username is found but the password verification fails, nil
+//     is returned immediately.
+//  2. If username and password are not both provided, the function attempts to
+//     match a "local" type identity using the userID.
+//
+// If no matching identity is found for the given inputs, nil is returned.
+func (s *State) IdentityFromInputs(userID *uint32, username, password string) *Identity {
 	s.reading()
 
-	for _, identity := range s.identities {
-		switch {
-		case identity.Local != nil && userID != nil:
-			if identity.Local.UserID == *userID {
+	switch {
```

Comment:
This seems a little strange, is there a reason we're using a `switch` here rather than an `if`?

Reply:
From my point of view, the benefit is if we add more types in the future, `switch` will be better. And, even if an identity has multiple types, we should return just one here.

Reply:
This is true, but do we know whether we'll add more types in the (reasonably) near future? Until that happens, this code will look a little odd, but either way seems fine